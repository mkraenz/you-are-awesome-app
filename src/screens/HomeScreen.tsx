import React, { FC } from "react";
import { StyleSheet, View } from "react-native";
import { Card, Paragraph, Title, useTheme } from "react-native-paper";
import { connect } from "react-redux";
import Layout from "../components/common/Layout";
import RefreshPostsView from "../components/RefreshPostsView";
import { Route } from "../navigation/Route";
import { IPostContent } from "../state/state/IPost";
import { MapStateToProps } from "../state/state/MapStateToProps";
import { FullTheme } from "../themes/theme";

const styles = StyleSheet.create({
    container: {
        width: "100%",
        height: "100%",
        flex: 1,
        justifyContent: "center",
        alignItems: "stretch",
    },
});

interface Props {
    post: IPostContent;
}

const HomeScreen: FC<Props> = ({ post }) => {
    const theme = useTheme() as FullTheme;
    const cardStyle = theme.dark
        ? {
              backgroundColor: theme.colors.accentedCard,
          }
        : { backgroundColor: theme.colors.primary };
    const { author, text, country } = post;
    return (
        <Layout route={Route.Home}>
            <RefreshPostsView>
                <View style={styles.container}>
                    <Card style={cardStyle}>
                        <Card.Content>
                            <Title style={{ color: "white" }}>{text}</Title>
                            <Paragraph style={{ color: "white" }}>
                                {author} from {country}
                            </Paragraph>
                        </Card.Content>
                    </Card>
                </View>
            </RefreshPostsView>
        </Layout>
    );
};

const mapStateToProps: MapStateToProps<Props> = (state) => ({
    post: state.posts.currentPost,
});

export default connect(mapStateToProps)(HomeScreen);
