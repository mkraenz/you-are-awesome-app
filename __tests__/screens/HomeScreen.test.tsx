import React from "react";
import "react-native";
import { DefaultTheme, Provider as PaperProvider } from "react-native-paper";
import { Provider } from "react-redux";
// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";
import createMockStore from "redux-mock-store";
import HomeScreen from "../../src/screens/HomeScreen";
import TestLocalizationProvider from "../helpers/TestLocalizationProvider";

it("renders correctly", () => {
    const store = createMockStore([])({
        posts: {
            currentPost: {
                id: 1,
                author: "my-author",
                text: "my-text",
                country: "my-country",
            },
        },
    });

    const tree = renderer
        .create(
            <Provider store={store}>
                <PaperProvider theme={DefaultTheme}>
                    <TestLocalizationProvider>
                        <HomeScreen />
                    </TestLocalizationProvider>
                </PaperProvider>
            </Provider>
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});

// TODO test interactivity
