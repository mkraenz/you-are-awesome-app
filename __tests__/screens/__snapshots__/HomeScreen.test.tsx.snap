// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "height": "100%",
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 4,
        "width": "100%",
      }
    }
  >
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Array [
          Object {
            "backgroundColor": "#6200ee",
            "elevation": 4,
            "zIndex": 0,
          },
          Object {
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 3,
              "width": 0,
            },
            "shadowOpacity": 0.24,
            "shadowRadius": 4,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#6200ee",
            "elevation": 0,
            "flexDirection": "row",
            "height": 56,
            "marginTop": 0,
            "paddingHorizontal": 4,
          }
        }
      >
        <View
          style={
            Object {
              "width": 48,
            }
          }
        />
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "flex": 1,
                "paddingHorizontal": 12,
              },
              Array [
                false,
                Object {
                  "alignItems": "center",
                },
                Object {
                  "alignItems": "center",
                  "display": "flex",
                },
              ],
            ]
          }
        >
          <Text
            accessibilityRole="header"
            accessibilityTraits="header"
            accessible={true}
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "fontSize": 17,
                  },
                  undefined,
                ],
              ]
            }
          >
            You are Awesome App!
          </Text>
        </View>
        <View
          style={
            Object {
              "width": 48,
            }
          }
        />
      </View>
    </RCTSafeAreaView>
    <View
      style={
        Object {
          "flex": 1,
          "paddingHorizontal": 16,
        }
      }
    >
      <RCTSafeAreaView
        emulateUnlessSupported={true}
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <RCTScrollView
          contentContainerStyle={
            Object {
              "flex": 1,
            }
          }
          refreshControl={
            <RefreshControlMock
              onRefresh={[Function]}
            />
          }
        >
          <RCTRefreshControl />
          <View>
            <View
              style={
                Object {
                  "alignItems": "stretch",
                  "flex": 1,
                  "height": "100%",
                  "justifyContent": "center",
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#6200ee",
                    "borderRadius": 4,
                    "elevation": 1,
                    "shadowColor": "#000000",
                    "shadowOffset": Object {
                      "height": 0.5,
                      "width": 0,
                    },
                    "shadowOpacity": 0.24,
                    "shadowRadius": 0.75,
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={false}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "flexGrow": 1,
                      "flexShrink": 1,
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "paddingHorizontal": 16,
                        },
                        Object {
                          "paddingVertical": 16,
                        },
                        undefined,
                      ]
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                            "fontFamily": "System",
                            "fontWeight": "400",
                            "textAlign": "left",
                          },
                          Array [
                            Object {
                              "color": "rgba(0, 0, 0, 0.87)",
                              "fontFamily": "System",
                              "fontWeight": "500",
                              "textAlign": "left",
                              "writingDirection": "ltr",
                            },
                            Array [
                              Object {
                                "fontSize": 20,
                                "letterSpacing": 0.15,
                                "lineHeight": 30,
                                "marginVertical": 2,
                              },
                              Object {
                                "color": "white",
                              },
                            ],
                            Array [
                              Object {
                                "fontSize": 20,
                                "letterSpacing": 0.15,
                                "lineHeight": 30,
                                "marginVertical": 2,
                              },
                              Object {
                                "color": "white",
                              },
                            ],
                          ],
                        ]
                      }
                    >
                      my-text
                    </Text>
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                            "fontFamily": "System",
                            "fontWeight": "400",
                            "textAlign": "left",
                          },
                          Array [
                            Object {
                              "color": "rgba(0, 0, 0, 0.87)",
                              "fontFamily": "System",
                              "fontWeight": "400",
                              "textAlign": "left",
                              "writingDirection": "ltr",
                            },
                            Array [
                              Object {
                                "fontSize": 14,
                                "letterSpacing": 0.25,
                                "lineHeight": 20,
                                "marginVertical": 2,
                              },
                              Object {
                                "color": "white",
                              },
                            ],
                            Array [
                              Object {
                                "fontSize": 14,
                                "letterSpacing": 0.25,
                                "lineHeight": 20,
                                "marginVertical": 2,
                              },
                              Object {
                                "color": "white",
                              },
                            ],
                          ],
                        ]
                      }
                    >
                      my-author
                       from 
                      my-country
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </RCTSafeAreaView>
    </View>
  </View>
</View>
`;
